pkg/upgrade/key_pair_test.go

```diff
@@ -81,70 +79,25 @@ func TestNewRestConfigFromCASecretRefError(t *testing.T) {
 	}
 }
 
-func TestNewRestConfigFromClusterField(t *testing.T) {
-	cluster := &v1alpha1.Cluster{
-		Spec: v1alpha1.ClusterSpec{
-			ProviderSpec: v1alpha1.ProviderSpec{
-				Value: &runtime.RawExtension{
-					Raw: []byte(fmt.Sprintf(`{"test": {"cert": "%s", "key": "%s"}}`, caCertificate, caPrivateKey)),
-				},
-			},
-		},
-	}
-	cfg, err := upgrade.NewRestConfigFromCAClusterField(cluster, "spec.providerSpec.value.test", "https://example.com:8888")
-	if err != nil {
-		t.Fatalf("%+v", err)
-	}
-	if cfg == nil {
-		t.Fatal("*rest.Config should not be nil")
-	}
-}
-
-func TestNewRestConfigFromClusterFieldErrors(t *testing.T) {
-	testcases := []struct {
-		name    string
-		cluster *v1alpha1.Cluster
-		field   string
-	}{
-		{
-			name: "an empty cluster object",
-		},
-		{
-			name:    "a bad field path",
-			cluster: &v1alpha1.Cluster{},
-			field:   "some field",
-		},
-	}
-	for _, tc := range testcases {
-		t.Run(tc.name, func(t *testing.T) {
-			cfg, err := upgrade.NewRestConfigFromCAClusterField(tc.cluster, tc.field, "https://example.com:8888")
-			if err == nil {
-				t.Fatal("expected an error but did not get one")
-			}
-			if cfg != nil {
-				t.Fatal("*rest.Config should always be nil")
-			}
-
-		})
-	}
-}
-
-func TestNewRestConfigFromKubeconfigSecretRef(t *testing.T) {
+/*func TestNewRestConfigFromKubeconfigSecretRef(t *testing.T) {
 	secret := &secrets{
 		secret: &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test",
+			},
 			Data: map[string][]byte{
 				"kubeconfig": []byte(kubeconfigBytes),
```

Comment:
I was playing around with this test -- if you change "kubeconfig" to "value" (or export `kubeconfigSecretKey` from the `upgrade` package and use that) this test can stay!